tests/integration_tests/test_interim_prod_r_3_1.py

```diff
@@ -0,0 +1,135 @@
+from scene_select.do_ard import ODC_FILTERED_FILE
+
+from util import (
+    get_list_from_file,
+    get_expected_file_paths,
+)
+
+METADATA_DIR = (
+    Path(__file__).parent.joinpath("..", "test_data", "odc_setup", "metadata").resolve()
+)
+METADATA_TYPES = [
+    os.path.join(METADATA_DIR, "eo3_landsat_l1.odc-type.yaml"),
+    os.path.join(METADATA_DIR, "eo3_landsat_ard.odc-type.yaml"),
+]
+
+PRODUCTS_DIR = (
+    Path(__file__).parent.joinpath("..", "test_data", "odc_setup", "eo3").resolve()
+)
+
+PRODUCTS = [
+    os.path.join(PRODUCTS_DIR, "l1_ls8.odc-product.yaml"),
+    os.path.join(PRODUCTS_DIR, "l1_ls8_c2.odc-product.yaml"),
+    os.path.join(PRODUCTS_DIR, "ard_ls8.odc-product.yaml"),
+]
+
+DATASETS_DIR = (
+    Path(__file__)
+    .parent.joinpath(
+        "..",
+        "test_data",
+        "integration_tests",
+    )
+    .resolve()
+)
+DATASETS = [
+    os.path.join(
+        DATASETS_DIR,
+        "c3/LC81020792023029_do_interim/LC08_L1GT_102079_20230129_20230227_02_T2.odc-metadata.yaml",
+    ),
+]
+
+pytestmark = pytest.mark.usefixtures("auto_odc_db")
+
+
+def test_interim_prod_r_3_1(tmp_path):
+    """
+    This is the collective test that implements the requirement as
+    defined at the top of this test suite.
+    """
+
+    # Observe that there is a date exclusion filter passed
+    # into the process
+    cmd_params = [
+        "--products",
+        '[ "usgs_ls8c_level1_2" ]',
+        "--logdir",
+        tmp_path,
+        "--allowed-codes",
+        "Australian_AOI_107069_added.json",
+        "--days-to-exclude",
+        '["2009-01-03:2009-01-05"]',
+        "--interim-days-wait",
+        5,
+    ]
+
+    runner = CliRunner()
+    result = runner.invoke(
+        scene_select,
+        args=cmd_params,
+    )
+
+    assert result.exit_code == 0, "The scene_select process failed to execute"
+
+    # Use glob to search for the log file
+    # within filter-jobid-* directories
+    matching_files = list(Path(tmp_path).glob("filter-jobid-*/" + GEN_LOG_FILE))
+
+    # There's only ever 1 copy of this file
+    assert (
+        matching_files and matching_files[0] is not None
+    ), f"Scene select failed. Log is not available - {matching_files}"
+
+    assert matching_files and matching_files[0] is not None, (
+        "Scene select failed. List of entries to process is not available -",
+        f" {matching_files}",
+    )
+
+    found_log_line = False
+    with open(matching_files[0], encoding="utf-8") as ard_log_file:
+        for line in ard_log_file:
+            try:
+                jline = json.loads(line)
+                print(f"Line is {line}")
+                if (
+                    all(key in jline for key in ("event", "landsat_scene_id", "level"))
+                    and jline["event"] == "No ancillary. Processing to interim"
+                    and jline["landsat_scene_id"]
+                    == "LC08_L1GT_102079_20230129_20230227_02_T2"
+                    and jline["level"] == "debug"
```

Comment:
Does jline["level"] == "debug"  need to be checked?

Reply:
good question. The reason why i did that was that scene select will only print that event message, `"No ancillary. Processing to interim"` under the debug level... I did consider not including this test.... Happpy to remove it if you feel we dont need to have it since the following test would query the list of ards to process .

Reply:
Thanks. Let's keep it.